File: php7/src/ChainOfResponsibility/Agent/BranchManager.php
<?php

declare(strict_types = 1);


namespace PetrePatrasc\ChainOfResponsibility\Agent;


use PetrePatrasc\ChainOfResponsibility\LoanApplicationRequest;
use PetrePatrasc\ChainOfResponsibility\LoanApplicationResponse;

class BranchManager extends AbstractLoanAgent
{
    public function __toString(): string
    {
        return 'Branch Manager';
    }

    /**
     * @inheritDoc
     */
    public function handleLoanAmount(LoanApplicationRequest $loanRequest): LoanApplicationResponse
    {
        if ($loanRequest->getAmount() < 10000) {
            return $this->approveLoanRequest();
        }

        return $this->getNextInChain()->handleLoanAmount($loanRequest);
    }

}

File: java/src/ro/petrepatrasc/chainofresponsability/agent/BranchManager.java
package ro.petrepatrasc.chainofresponsability.agent;

import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationRequest;
import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationResponse;

/**
 * @author Petre Pătrașc <petre@dreamlabs.ro>
 */
public class BranchManager extends AbstractLoanAgent {

    public static final double AGENT_THRESHOLD = 10000;

    @Override
    public LoanApplicationResponse handleLoanRequest(LoanApplicationRequest loanRequest) {
        if (loanRequest.getAmount() < this.AGENT_THRESHOLD) {
            return this.approveLoanRequest();
        }

        return this.getNextInChain().handleLoanRequest(loanRequest);
    }
}
